mobile/src/main/java/ch/epfl/sdp/mobile/application/Profile.kt

```diff
@@ -15,6 +15,9 @@ interface Profile {
   /** The background color associated to the [Profile]. */
   val backgroundColor: Color
 
+  /** The unique identifier of the person. */
+  val uid: String
```

Comment:
Adding a `uid` to the `Person` interface is an abstraction leak. You should prefer using some generic composable functions, where the ViewModel layer is responsible to provide a proper type which takes care of managing `Person` identity, or other fields which may be relevant.

This is the pattern we use in the `ChessBoard` composable : https://github.com/epfl-SDP/android/blob/main/mobile/src/main/java/ch/epfl/sdp/mobile/ui/game/ChessBoard.kt#L43-L47. Notice how `ChessBoardState` has a generic `Piece` https://github.com/epfl-SDP/android/blob/main/mobile/src/main/java/ch/epfl/sdp/mobile/ui/game/ChessBoardState.kt#L21, and how it is the ViewModel which takes care of managing the identifiers of pieces with the dedicated `SnapshotPiece` type : https://github.com/epfl-SDP/android/blob/main/mobile/src/main/java/ch/epfl/sdp/mobile/state/StatefulGameScreen.kt#L28-L32.

This pattern comes up quite often in Compose - this is kind of similar to the generic composables `LazyColumn` or `LazyRow`.